net-libs/libbitcoinconsensus/libbitcoinconsensus-0.15.1.recipe

```diff
@@ -0,0 +1,98 @@
+SUMMARY="Original Bitcoin client"
+DESCRIPTION="BitCoin Core is the backbone of the network. It downloads and, \
+by default, stores the entire history of Bitcoin transactions (which is \
+currently more than 100 GBs); depending on the speed of your computer and \
+network connection, the synchronization process can take anywhere from a few \
+hours to a day or more."
+HOMEPAGE="https://bitcoincore.org/"
+COPYRIGHT="2009-2017, Bitcoin Core Developers"
+LICENSE="MIT"
+REVISION="1"
+SOURCE_URI="https://github.com/bitcoin/bitcoin/archive/v$portVersion.tar.gz"
+CHECKSUM_SHA256="98b3a98a67054123bef6714e4035e18f9250a1b6d63829258ad057906e75d8b1"
+SOURCE_DIR="bitcoin-$portVersion"
+
+ARCHITECTURES="!x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
```

Comment:
```diff
-ARCHITECTURES="!x86_gcc2 x86 x86_64"
+ARCHITECTURES="!x86_gcc2 ?x86 ?x86_64"
-SECONDARY_ARCHITECTURES="x86"
+SECONDARY_ARCHITECTURES="?x86"
```
Shall we mark this as untested and merge this PR?

Reply:
Sounds ok to me, for now there won't be any progress untill some boost expert comes around :)

Reply:
missing sleep funcion is comming from boost_threads I think, and since it's (still) not found hence the error (my 2 cents)
Also noticed pthread/lpthread is not found by configue, maybe related?

Reply:
From configure.ac "sleep was removed after boost 1.56"

Reply:
That's interesting. Maybe the absence of conditions on the versions of the boost libraries is allowing a mix from different versions on packages that may be co-activated.
So maybe adding some conditions (to force either boost < 1.56 or boost >= 1.56) on the required versions for all boost packages could help. (I'm not trying this, I gave up boost for the moment :)

Reply:
Looks like they are working on it too: https://github.com/bitcoin/bitcoin/pull/11046

Reply:
Got it building, but with some hacks, now running into:

> boot/system/develop/headers/x86/boost/move/unique_ptr.hpp:399:4: error: constexpr constructor does not have empty body

(think I saw this passing by also for OpenClonk) ...

Reply:
So many errors... with Boost.